File: src/main/java/com/hjj/apiserver/repositroy/CardRepository.java
package com.hjj.apiserver.repositroy;

import com.hjj.apiserver.domain.CardEntityJava;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface CardRepository extends JpaRepository<CardEntityJava, Long> {
    List<CardEntityJava> findByUserEntity_UserNoAndDeleteYn(Long userNo, char deleteYn);
    Optional<CardEntityJava> findByCardNoAndUserEntity_UserNo(Long cardNo, Long userNo);
}

File: src/main/kotlin/com/hjj/apiserver/repository/card/CardRepository.kt
package com.hjj.apiserver.repository.card

import com.hjj.apiserver.domain.card.Card
import org.springframework.data.jpa.repository.JpaRepository

interface CardRepository: JpaRepository<Card, Long> {
    fun findByUserEntity_UserNoAndIsDeleteIsFalse(userNo: Long): MutableList<Card>
    fun findByCardNoAndUserEntity_UserNoAndIsDeleteIsFalse(cardNo: Long, userNo: Long): Card?
    fun findByUserEntity_UserNo(userNo: Long): MutableList<Card>
}